src/wallet/wallet.cpp

```diff
@@ -1476,11 +1476,13 @@ void CWallet::TransactionAddedToMempool(const CTransactionRef& ptx, int64_t nAcc
     }
 }
 
-void CWallet::TransactionRemovedFromMempool(const CTransactionRef &ptx) {
-    LOCK(cs_wallet);
-    auto it = mapWallet.find(ptx->GetHash());
-    if (it != mapWallet.end()) {
-        it->second.fInMempool = false;
+void CWallet::TransactionRemovedFromMempool(const CTransactionRef &ptx, MemPoolRemovalReason reason) {
+    if (reason != MemPoolRemovalReason::CONFLICT) {
+        LOCK(cs_wallet);
+        auto it = mapWallet.find(ptx->GetHash());
+        if (it != mapWallet.end()) {
+            it->second.fInMempool = false;
+        }
```

Comment:
Why only do this if it's not conflict?

Reply:
It's to preserve the original logic in `MempoolEntryRemoved` https://github.com/dashpay/dash/pull/4160/files#diff-cd341be4887e632ceba211aee03c948d964396f7aed0146435163844fbcbad42L169